numpy/_core/src/npysort/quicksort.cpp

```diff
@@ -81,18 +81,23 @@ inline bool quicksort_dispatch(T *start, npy_intp num)
     void (*dispfunc)(TF*, intptr_t) = nullptr;
     if (sizeof(T) == sizeof(uint16_t)) {
         #ifndef NPY_DISABLE_OPTIMIZATION
-            #include "simd_qsort_16bit.dispatch.h"
+            #include "x86_simd_qsort_16bit.dispatch.h"
         #endif
         NPY_CPU_DISPATCH_CALL_XB(dispfunc = np::qsort_simd::template QSort, <TF>);
     }
-    #if !DISABLE_HIGHWAY_OPTIMIZATION
     else if (sizeof(T) == sizeof(uint32_t) || sizeof(T) == sizeof(uint64_t)) {
         #ifndef NPY_DISABLE_OPTIMIZATION
-            #include "simd_qsort.dispatch.h"
+            #include "x86_simd_qsort.dispatch.h"
         #endif
         NPY_CPU_DISPATCH_CALL_XB(dispfunc = np::qsort_simd::template QSort, <TF>);
+        // If not dispatched for avx-512 or avx2:
+        if (dispfunc == nullptr) {
```

Comment:
Could we do this at compile time by checking the CPU macros?  Something like:

```
#ifdef NPY_CPU_X86
... quicksort checks for x86 ...
#else
.. quicksort checks for aarch64 ...
#endif
````

Unsure if the compiler would pick up on `dispfunc` always being null for aarch64 🤔 .

Reply:
Makes sense.